src/Operations.jl

```diff
@@ -325,13 +403,41 @@ function update_manifest(env::EnvCache, uuid::UUID, name::String, hash::SHA1, ve
         push!(infos, info)
     end
     info["version"] = string(version)
-    info["hash-sha1"] = string(hash)
+    if isempty(path)
+        @assert hash != nothing
+        haskey(info, "path") && delete!(info, "path")
+        info["hash-sha1"] = string(hash)
+    else
+        haskey(info, "hash-sha1") && delete!(info, "hash-sha1")
+        info["path"] = path
+    end
     delete!(info, "deps")
-    for path in registered_paths(env, uuid)
-        data = load_package_data(UUID, joinpath(path, "dependencies.toml"), version)
-        data == nothing && continue
-        info["deps"] = convert(Dict{String,String}, data)
-        break
+    if !isempty(path)
+        reqfile = joinpath(path, "REQUIRE")
+
+        !isfile(reqfile) && return
+        deps = Dict{String, String}()
+        pkgs = PackageSpec[]
+        for r in filter!(r->r isa Pkg2.Reqs.Requirement, Pkg2.Reqs.read(reqfile))
+            push!(pkgs, PackageSpec(r.package))
+        end
+        # TODO: Get rid of this one?
```

Comment:
This is quite inefficient. When everything has been moved to Pkg3, the UUID would be available from the `Dependencies.toml` or whatever it is called, and this look up wouldn't be needed.